ui/src/warp_runner/ui_adapter/mod.rs

```diff
@@ -0,0 +1,156 @@
+//! an event from Warp isn't necessarily what the UI needs to display. and the UI doesn't have access to RayGun, MultiPass, ect. As a result,
+//! a translation must be performed by WarpRunner.
+//!
+
+use uuid::Uuid;
+use warp::{
+    crypto::DID,
+    multipass::MultiPassEventKind,
+    raygun::{Conversation, MessageOptions, RayGunEventKind},
+};
+
+use crate::state::{self, chats};
+
+#[allow(clippy::large_enum_variant)]
+pub enum RayGunEvent {
+    ConversationCreated(state::Chat),
+    ConversationDeleted(Uuid),
+}
+
+pub enum MultiPassEvent {
+    None,
+    FriendRequestReceived(state::Identity),
+    FriendRequestSent(state::Identity),
+    FriendAdded(state::Identity),
+    FriendRemoved(state::Identity),
+    FriendRequestCancelled(state::Identity),
+    FriendOnline(state::Identity),
+    FriendOffline(state::Identity),
+}
+
+pub async fn did_to_identity(did: DID, account: &mut super::Account) -> state::Identity {
+    let warp_identity = account
+        .get_identity(did.into())
+        .await
+        .expect("could not get warp identity");
+    state::Identity::from(
+        warp_identity
+            .first()
+            .cloned()
+            .expect("could not get warp identity"),
```

Comment:
Should use `ok_or(Error::IdentityDoesntExist)` if `Result` is used on the function.